torch/ao/quantization/backend_config/qnnpack.py

```diff
@@ -11,63 +11,112 @@
     _get_rnn_op_configs,
     _get_share_qparams_op_configs,
 )
-from .backend_config import BackendConfig, DTypeConfig
+from .backend_config import BackendConfig, DTypeConfig, DTypeWithConstraints
 
 
 # ===================
 # |  DTYPE CONFIGS  |
 # ===================
 
-# TODO: For now, these DTypeConfigs are identical to the ones defined in native.py
-# In the future, once we support specifying quant_min/quant_max and scale_min/scale_max,
-# these will diverge. In particular, for QNNPACK, we will restrict the weight quantized
-# values to within [-127, 127] and set the min scale value to 2 ** -12.
+def _get_activation_dtype_with_constraints(dtype: torch.dtype):
+    return DTypeWithConstraints(dtype=dtype, scale_min_lower_bound=2 ** -12)
 
-qnnpack_weighted_op_int8_dtype_config = DTypeConfig(
-    input_dtype=torch.quint8,
-    output_dtype=torch.quint8,
-    weight_dtype=torch.qint8,
+def _get_weight_dtype_with_constraints(dtype: torch.dtype):
+    return DTypeWithConstraints(
+        dtype=dtype,
+        quant_min_lower_bound=-127,
+        quant_max_upper_bound=127,
+        scale_min_lower_bound=2 ** -12
+    )
+
+act_qint8_with_constraints = _get_activation_dtype_with_constraints(torch.qint8)
+act_quint8_with_constraints = _get_activation_dtype_with_constraints(torch.quint8)
+act_fp16_with_constraints = _get_activation_dtype_with_constraints(torch.float16)
+act_fp_with_constraints = _get_activation_dtype_with_constraints(torch.float)
+weight_qint8_with_constraints = _get_weight_dtype_with_constraints(torch.qint8)
+weight_quint8_with_constraints = _get_weight_dtype_with_constraints(torch.quint8)
+weight_quint4x2_with_constraints = _get_weight_dtype_with_constraints(torch.quint4x2)
+weight_fp16_with_constraints = _get_weight_dtype_with_constraints(torch.float16)
+
+# weighted op
+
+qnnpack_weighted_op_qint8_dtype_config = DTypeConfig(
+    input_dtype=act_qint8_with_constraints,
+    output_dtype=act_qint8_with_constraints,
```

Comment:
These qint8 `DTypeConfig`s are added to work with `default_symmetric_qnnpack_qconfig` and its per_channel and QAT equivalents

Reply:
I'm wondering if we want to add `qscheme` support to this DTypeConfig as well

Reply:
I see, you mean require this one to be `per_tensor_symmetric` or `per_channel_symmetric`?

Reply:
Would we want to support affine qint8 weight with affine qint8 activations in the future? (right now we have symmetric qint8 weight with affine qint8 activations) @digantdesai

Reply:
> I see, you mean require this one to be per_tensor_symmetric or per_channel_symmetric?

yeah. this should be a separate PR though